TestingHomework.Tests/ProductAdminManagerTests.cs

```diff
@@ -1,108 +1,146 @@
+            //// act
+            Product actualProduct = manager.GetProduct(expectedProduct.Id);
+            ////assert
+            expectedProduct.ShouldDeepEqual(actualProduct);
+        }
+        
+        [Test]
+        public void GetAllProducts()
+        {
+            // arrange
+            int expectedItemCount = 5;
+            IEnumerable<Product> expectedProducts = dataPrep.Products.CreateManyForList(expectedItemCount);
+
+            mockContainer.Arrange<IProductAccessor>(accessor => accessor.GetAllProducts()).Returns(expectedProducts);
 
-            var actualProduct = manager.GetProduct(savedProduct.Id);
-            Assert.Equal(savedProduct.Id, actualProduct.Id);
-            Assert.Equal(savedProduct.Description, actualProduct.Description);
-            Assert.Equal(savedProduct.Price, actualProduct.Price);
+            //// act
+            IEnumerable<Product> actualProductList = manager.GetAllProducts();
 
+            ////assert
+            Assert.AreEqual(expectedProducts.Count(), actualProductList.Count());
+            expectedProducts.ShouldDeepEqual(actualProductList);
         }
 
-        [Fact]
+        [Test]
         public void SaveProduct_NewProduct()
         {
-            var savedProduct = manager.SaveProduct(new Product()
-            {
-                Description = "asdfas",
-                Price = 10.50
-            });
+            // arrange
+            Product expectedProduct = dataPrep.Products.CreateData(isPersisted: false);
+
+            mockContainer.Arrange<IProductAccessor>(accessor => accessor.SaveProduct(expectedProduct)).Returns(new SaveResult<Product>(expectedProduct));
 
-            Assert.False(savedProduct.Id == Guid.Empty);
-            Assert.Equal("asdfas", savedProduct.Description);
-            Assert.Equal(10.50, savedProduct.Price);
+            // act
+            SaveResult<Product> actualProductResult = manager.SaveProduct(expectedProduct);
+            Product actualProduct = actualProductResult.Result;
+
+            //assert
+            //Assert.False(actualProduct.Id == Guid.Empty);
+            Assert.IsTrue(actualProductResult.Success);
+            expectedProduct.WithDeepEqual(actualProduct).IgnoreSourceProperty((p) => p.Id);
         }
 
-        [Fact]
+        [Test]
         public void SaveProduct_UpdateProduct()
         {
-            var existingProduct = manager.SaveProduct(new Product()
-            {
-                Description = "asdfas",
-                Price = 10.50
-            });
+            // arrange
+            Product expectedProduct = dataPrep.Products.CreateData();
+            expectedProduct.Description = Guid.NewGuid().ToString();
 
-            existingProduct.Price = 1.00;
+            mockContainer.Arrange<IProductAccessor>(accessor => accessor.SaveProduct(expectedProduct)).Returns(new SaveResult<Product>(expectedProduct));
```

Comment:
These mocks don't currently matter since you only run your tests as integration.
With the current state of the managers, its true that non-integration tests would be pretty meaningless. 
I think that is a significant warning sign. If you get little benefit from mocking in a good portion of tests, then
there isn't enough separation of responsibilities in the code